src/dotty/tools/dotc/transform/Erasure.scala

```diff
@@ -319,7 +355,7 @@ object Erasure extends TypeTestsCasts{
           assert(sym.isConstructor, s"${sym.showLocated}")
           select(qual, defn.ObjectClass.info.decl(sym.name).symbol)
         }
-        else if (qualIsPrimitive && !symIsPrimitive || qual.tpe.isErasedValueType)
+        else if (qualIsPrimitive && !symIsPrimitive || qual.tpe.widen.isErasedValueType)
```

Comment:
I'm never really sure of when I should use `widen` and when I should use `widenDealias`, but I'll trust you on that one :).